api/handlers/capturesource.go

```diff
@@ -212,3 +224,72 @@ func CreateCaptureSource(ctx *fiber.Ctx) error {
 		ID: &id,
 	})
 }
+
+// UpdateCaptureSource updates a capture source.
+func UpdateCaptureSource(ctx *fiber.Ctx) error {
+	// Parse URL.
+	id, err := strconv.ParseInt(ctx.Params("id"), 10, 64)
+	if err != nil {
+		return api.InvalidRequestURL(ctx)
+	}
+
+	// Parse body.
+	var body UpdateCaptureSourceBody
+	if ctx.BodyParser(&body) != nil {
+		return api.InvalidRequestJSON(ctx)
+	}
+
+	// Check that capture source has no video streams associated with it.
+	// TODO: Implement this.
+
+	// Update data in the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("CaptureSource", id)
+	var captureSource model.CaptureSource
+
+	if store.Update(context.Background(), key, func(e datastore.Entity) {
```

Comment:
Can the call to store.Update happen outside of the if condition?

Reply:
fixed